clang/test/CodeGenSYCL/anonymous_integration_footer.cpp

```diff
@@ -0,0 +1,363 @@
+// RUN: %clang_cc1 -fsycl-is-device -std=c++17 -triple spir64-unknown-unknown-sycldevice -fsycl-int-footer=%t.h %s -emit-llvm -o %t.ll
+// RUN: FileCheck -input-file=%t.h %s
+// A test that validates the more complex cases of the specialization-constant
+// integration footer details, basically any situation we can come up with that
+// has an anonymous namespace.
+
+#include "Inputs/sycl.hpp"
+int main() {
+  cl::sycl::kernel_single_task<class first_kernel>([]() {});
+}
+
+using namespace cl;
+// Example ways in which the application can declare a "specialization_id"
+// variable.
+struct S1 {
+  static constexpr sycl::specialization_id a{1};
+  // CHECK: namespace sycl {
+  // CHECK-NEXT: namespace detail {
+  // CHECK-NEXT: template<>
+  // CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::S1::a>() {
+  // CHECK-NEXT: return "";
+  // CHECK-NEXT: }
+  // CHECK-NEXT: } // namespace detail
+  // CHECK-NEXT: } // namespace sycl
+};
+constexpr sycl::specialization_id b{2};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::b>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+inline constexpr sycl::specialization_id c{3};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::c>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+static constexpr sycl::specialization_id d{4};
+// CHECK-NEXT: namespace sycl {
+// CHECK-NEXT: namespace detail {
+// CHECK-NEXT: template<>
+// CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::d>() {
+// CHECK-NEXT: return "";
+// CHECK-NEXT: }
+// CHECK-NEXT: } // namespace detail
+// CHECK-NEXT: } // namespace sycl
+
+namespace {
+struct S2 {
+  static constexpr sycl::specialization_id a{18};
+  // CHECK-NEXT: namespace {
+  // CHECK-NEXT: namespace __sycl_detail {
+  // CHECK-NEXT: static constexpr decltype(S2::a) &__spec_id_shim_[[SHIM_ID:[0-9]+]]() {
+  // CHECK-NEXT: return S2::a;
+  // CHECK-NEXT: }
+  // CHECK-NEXT: } // namespace __sycl_detail
+  // CHECK-NEXT: } // namespace
+  // CHECK-NEXT: namespace sycl {
+  // CHECK-NEXT: namespace detail {
+  // CHECK-NEXT: template<>
+  // CHECK-NEXT: inline const char *get_spec_constant_symbolic_ID<::__sycl_detail::__spec_id_shim_[[SHIM_ID]]()>() {
+  // CHECK-NEXT: return ::__sycl_detail::__spec_id_shim_[[SHIM_ID]]();
```

Comment:
It doesn't seem like this code will compile: we return a pointer to a `specialization_id` variable from a function which is expected to return `const char *`.  `symbolic_ID` should be return there